zabbix_auto_config/processing.py

```diff
@@ -666,11 +677,27 @@ def do_update(self):
             # if api ver >= 4.4
             if "inventory_mode" in zabbix_host.keys():
                 if zabbix_host["inventory_mode"] != "1":
-                    self.set_inventory(zabbix_host, 1)
+                    self.set_inventory_mode(zabbix_host, 1)
             # else api ver. < 4.4
             else:
                 if zabbix_host["inventory"]["inventory_mode"] != "1":
-                    self.set_inventory(zabbix_host, 1)
+                    self.set_inventory_mode(zabbix_host, 1)
+
+            if db_host["inventory"]:
```

Comment:
This is always true? If so: It should be removed.

Reply:
Actually. It depends. I'm not sure what the intended behavior is if you remove the last inventory item? They won't be removed now, as far as I can tell.

Reply:
when the last inventory is removed, it will not be removed from Zabbix as you state. When managed_inventory is set to null in configuration file, one loses basis data needed to query zabbix inventory, compare and distinguish inventories managed by zac and users via zabbix UI. 
This case is discussed internally (USIT/DIA) and it is considered as a rare case and will be handled separately. 
A solution to this case would be to store managed_inventory to permanent storage e.g. db or file as well and compare it with the current managed_inventory from configuration file but it is not implemented since it might complicate the code for a rare case.